autoload/codefmt/formatterhelpers.vim

```diff
@@ -0,0 +1,59 @@
+" Copyright 2017 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+""
+" @public
+" Format lines in the current buffer via a formatter invoked by {cmd}. The
+" command includes the explicit range line numbers to use, if any.
+"
+" @throws ShellError if the {cmd} system call fails
+function! codefmt#formatterhelpers#Format(cmd) abort
```

Comment:
Just wondering about how we might extend this for #60, if we wanted to centralize error-handling later.

Possibly we could add an extra optional argument that extracted the line:column:message from any error text, and caused this to handle adding errors to the quickfix list?

Reply:
AFAICT this accommodates it just fine if we provide another helper to accept the extracted errors (and let the caller handle the details of extracting the errors, which will vary case-by-case and won't generalize well).